runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5061,13 +5061,57 @@ TR_J9ByteCodeIlGenerator::loadAuto(TR::DataType type, int32_t slot, bool isAdjun
    push(load);
    }
 
+static bool isFieldResovled(TR::Compilation *comp, TR_ResolvedJ9Method * owningMethod, int32_t cpIndex, bool isStore)
+   {
+   uint32_t offset = 0;
+   TR::DataType type = TR::NoType;
+   bool isVolatile = true, isFinal = false, isPrivate = false, isUnresolvedInCP;
+   bool bResolved = owningMethod->fieldAttributes(comp, cpIndex, &offset, &type, &isVolatile, &isFinal,
+                                    &isPrivate, isStore, &isUnresolvedInCP, true /* needsAOTValidation */);
+   return bResolved;
+   }
+
 void
 TR_J9ByteCodeIlGenerator::loadInstance(int32_t cpIndex)
    {
    if (_generateReadBarriersForFieldWatch && comp()->compileRelocatableCode())
       comp()->failCompilation<J9::AOTNoSupportForAOTFailure>("NO support for AOT in field watch");
-   TR::SymbolReference * symRef = symRefTab()->findOrCreateShadowSymbol(_methodSymbol, cpIndex, false);
-   loadInstance(symRef);
+
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+   if (TR::Compiler->om.areValueTypesEnabled() &&
+       owningMethod->isFieldQType(cpIndex))
```

Comment:
Can the `isFieldQType` really be called before checking if it's resolved?

Reply:
`isFieldQType()` essentially calls VM to check `J9ROMNameAndSignature` as below and see if it contains `Q` keyword. My understanding is that it doesn't require the field to be resolved. Or does it need to? 

https://github.com/eclipse/openj9/blob/fef3f947a0086af748f59f5aa688b09960a43421/runtime/vm/ValueTypeHelpers.hpp#L248-L261